homeassistant/components/syncthru/sensor.py

```diff
@@ -150,16 +157,21 @@ def device_state_attributes(self):
 
 
 class SyncThruMainSensor(SyncThruSensor):
-    """Implementation of the main sensor, monitoring the general state."""
+    """Implementation of the main sensor, conducting the actual polling."""
 
     def __init__(self, syncthru, name):
         """Initialize the sensor."""
         super().__init__(syncthru, name)
         self._id_suffix = '_main'
 
-    def update(self):
+    async def async_update(self):
         """Get the latest data from SyncThru and update the state."""
-        self.syncthru.update()
+        try:
+            await self.syncthru.update()
+        except ValueError:
+            # if an exception is thrown, printer does not support syncthru
+            _LOGGER.info("Samsung printer at %s does not support SyncThru",
```

Comment:
Won't this spam the log every 30 seconds update is called for non supported printers? That's not a good experience.

It would be better to not add those printers at all. Can't we check this also for the manually configured printers and fail set up if support is lacking?

Reply:
Could we remove the entity if this error happens and ask the user to change the config?

There's no point in keeping the entity if this error happens, right?

Reply:
That's right, there is no point in keeping the entity. Yet I have no idea about how to remove an entity that was added once before. Upcoming updates could be skipped and the logging only issued once, if you have got better ideas/knowledge of the documentation please let me know :)